src/DynamoCore/Configuration/DebugModes.cs

```diff
@@ -35,32 +34,30 @@ public class DebugMode
             public bool IsEnabled;
         }
 
-        private static void AddDebugMode(string name, string description)
+        private static void AddDebugMode(string name, string description, bool isEnabled = false)
         {
             debugModes[name] = new DebugMode()
             {
                 Name = name,
                 Description = description,
-                IsEnabled = false
+                IsEnabled = isEnabled
             };
         }
         private static void RegisterDebugModes()
         {
             // Register app wide new debug modes here.
-            AddDebugMode("Python3DebugMode", "Enable/disable Python3 Engine.");
+            AddDebugMode("Python3DebugMode", "Enable/disable Python3 Engine.", true);
```

Comment:
Seems after this change, even without a debug.config, debug modes can be turned on purely with the API, is that the intention here?

Reply:
Yes. I thought it would be better not to require any file for enabling the Python engine selection, and have it turned on by default.

Reply:
- Do users' disable debug mode setting still get respected?
- This way we will have two source of truth.. Which we can discuss within the team, I understand it maybe annoying to setup debug.config locally but in general I would support keep only one source, and use as least debug modes as possible.
- I feel this also opens up the door for extensions to touch debug modes potentially

Please give some consideration.

Reply:
Hi Aaron. In this case we are only allowing to set the default to something other than `false`. Other than that I didn't make any changes.

Of course a debug mode can still be affected by a loaded XML, that's no different from before. The fact that it was `false` by default didn't mean it didn't exist. We always had "two sources of truth" if you may: a default and a config override.

I don't see how an extension could change these settings, without resorting to reflection. In any case, I only changed the visibility of one function to internal, so the door is not any more open than before.

Reply:
hmm but previously we are not allowing to set the default to enabled without a config correct? So extensions will not be able to do that either.

Reply:
Cool. Thanks for the discussion. I think it is safe to proceed!